src/wing/CTiglWingBuilder.cpp

```diff
@@ -253,28 +319,63 @@ CTiglWingBuilder::operator PNamedShape()
     return BuildShape();
 }
 // Set the name of each wing face
-void CTiglWingBuilder::SetFaceTraits (const std::string& wingUID, PNamedShape loft, bool hasBluntTE)
+void CTiglWingBuilder::SetFaceTraits (const std::vector<double>& guideCurveParams, const std::string& shapeUID, PNamedShape shape, bool hasBluntTE)
 {
-    unsigned int nSegments = _wing.GetSegmentCount();
+    auto params = guideCurveParams;
+    assert(std::is_sorted(std::begin(params), std::end(params)));
+
+    bool hasGuideCurves = params.size() > 0.;
+
+    size_t nFacesPerSegment = 2; // Without trailing edge
+    size_t idx_leading_edge = 1;
+    if (hasGuideCurves) {
+        if (!contains(params, -1.)) {
```

Comment:
Again, no big deal: But since you already assert that the params vector is sorted, you could check the first element of params against -1 instead of searching the whole vector.

Reply:
what, if the first element is e.g. -2? By accident?

Reply:
Ok, so say the first parameter is -2 by accident and -1 is not contained in params. Wouldn't prefixing params with -1 really screw up the parameter range then? The parameters would be eg [-1, -2, -0.5, 0.5, 2].

Reply:
Yes, you are right!